core/src/test/java/kafka/security/JaasTestUtils.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.security;
+
+import kafka.utils.TestUtils;
+
+import org.apache.kafka.clients.admin.ScramMechanism;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.utils.Java;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.apache.kafka.common.config.SaslConfigs.GSSAPI_MECHANISM;
+import static org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.OAUTHBEARER_MECHANISM;
+import static org.apache.kafka.common.security.plain.internals.PlainSaslServer.PLAIN_MECHANISM;
+
+public class JaasTestUtils {
+    public static class JaasSection {
+        private final String contextName;
+        private final List<JaasModule> modules;
+
+        public JaasSection(String contextName, List<JaasModule> modules) {
+            this.contextName = contextName;
+            this.modules = modules;
+        }
+
+        public List<JaasModule> getModules() {
+            return modules;
+        }
+
+        public String getContextName() {
+            return contextName;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("%s {\n  %s\n};\n",
+                    contextName,
+                    String.join("\n  ", modules.stream().map(Object::toString).toArray(String[]::new)));
```

Comment:
`modules.stream().map(Object::toString).collect(Collectors.joining("\n  "))`

Reply:
Hi @chia7712 
I have test the output in my local machine, it seems that the suggestions would have different out compare to original one.

Before:
```
KafkaServer {
  org.apache.kafka.common.security.plain.PlainLoginModule required
  debug=false
  password=""
  username="plain-admin";

};
```

After:
```
org.apache.kafka.common.security.plain.PlainLoginModule required
  debug=false
  password=""
  username="plain-admin";
```

Reply:
Pardon me, could you share the test code to me? I don't notice the format like "KafkaServer { .. " on my local

Reply:
The tests which extend `EndToEndAuthorizationTest` will fail due to error config, it seems that these tests utilize the `toString` method to generate configuration

Reply:
> The tests which extend EndToEndAuthorizationTest will fail due to error config, 

which one? I run a part of tests, and they pass with following changes.

```java
        @Override
        public String toString() {
            return String.format("%s {\n  %s\n};\n",
                    contextName,
                    modules.stream().map(Object::toString).collect(Collectors.joining("\n  ")));
        }
```

Reply:
I have misunderstanding of the comment, it works now. Sorry for that 😿